src/Nox.Lib/Factories/NoxTypeFactoryBase.cs

```diff
@@ -15,8 +15,9 @@ public NoxTypeFactoryBase(NoxSolution solution)
         public virtual T? CreateNoxType(Entity entityDefinition, string propertyName, dynamic? value)
         {
```

Comment:
we should be explicitly and have a method IsKey(keyname) and according to it get the right definition

Reply:
Added IsKey and updated the codebase accordingly